server/models/node.js

```diff
@@ -1,62 +1,85 @@
 /* vim: set sw=2 ts=2 expandtab : */
 const db = require('../db/db');
 
-async function getOne(chain_id, node_id) {
+async function getOne(chain_id, val_addr, from, to) {
   return new Promise(function(resolve, reject) {
     var query_str = "SELECT \
-        n.`node_id`, n.`moniker`, \
-        INET_NTOA(n.`ip_addr`) `ip_addr`, i.`n_peers`, i.`val_addr`, \
-        i.`latest_block_height`, i.`latest_block_time`, \
-        i.`catching_up`, i.`elapsed` \
-      FROM \
-        `nodes` n, `node_info` i \
-        JOIN ( \
-          SELECT \
-            `chain_id`, `node_id`, \
-            MAX(`latest_block_height`) `latest_block_height` \
-          FROM `node_info` \
-          WHERE `chain_id` = ? AND `node_id` = ? \
-          GROUP BY `node_id` \
-        ) m ON i.`chain_id` = m.`chain_id` \
-        AND i.`node_id` = m.`node_id` \
-        AND i.`latest_block_height` = m.`latest_block_height` \
-      WHERE n.`chain_id` = m.`chain_id` AND n.`node_id` = m.`node_id`";
-    query_var = [chain_id, node_id];
-    val = {};
-    db.query(query_str, query_var, function(err, rows, fields) {
+    n.`node_id`, n.`moniker`, ni.`val_addr`, \
+    ni.`catching_up`, ni.`elapsed`, ni.`timestamp`, \
+    ni.`latest_block_height`, ni.`latest_block_time`, \
+    ut.`uptime` \
+  FROM `nodes` n \
+  JOIN `node_info` ni \
+    ON n.`chain_id` = ni.`chain_id` \
+    AND n.`node_id` = ni.`node_id` \
+  JOIN ( \
+    SELECT o.`node_id`, COUNT(o.`timestamp`)/g.`count_t`*100 `uptime` \
+    FROM ( \
+      SELECT `node_id`, `timestamp` \
+      FROM `node_info` \
+      WHERE `chain_id` = ? AND `val_addr` = ? AND `timestamp` BETWEEN ? AND ? \
+    ) o \
+    JOIN ( \
+      SELECT COUNT(`timestamp`) `count_t` \
+      FROM `node_info` \
+      WHERE `chain_id` = 'ghost_chain_id' AND `timestamp` BETWEEN ? AND ? \
+    ) g \
+    GROUP BY o.`node_id` \
+  ) ut \
+  ON n.`node_id` = ut.`node_id` \
+  JOIN ( \
+    SELECT `node_id`, MAX(`timestamp`) `max_timestamp` \
+    FROM `node_info` \
+    WHERE `chain_id` = ? AND `val_addr` = ? AND `timestamp` BETWEEN ? AND ? \
+    GROUP BY `node_id` \
+  ) mt \
+  ON n.`node_id` = mt.`node_id` AND ni.`timestamp` = mt.`max_timestamp` \
+  ORDER BY n.`moniker`";
+    query_var = [chain_id, val_addr, from, to, from, to, chain_id, val_addr, from, to];
+    db.query(query_str, query_var, function(err, row, fields) {
```

Comment:
Too complex query. I don't get what it means. leave a comment explaining this query.